Create List, Destroy, Provider and createReducer documentation (#39)
SeasonedSoftware/croods: src/docs/List.mdx

---
name: List
route: /list
---

import { Playground } from 'docz'

# List

> The List component provides a way to fetch a list from an api, using the response as needed, fetching or set a resource and show or use the response data.

Reviewer: ```suggestion
The List component provides a way to fetch a collection of entries from an API's resource. Think of it as the `index/find` method from backend frameworks.
```

## API

| Property      | Type          | Required | Default    | Description                                                                                             |
| ------------- | ------------- | -------- | ---------- | ------------------------------------------------------------------------------------------------------- |

> | name          | String        | true     | -          | It defines which reducer will be used and, if no path have been provider, will set path to baseUrl/name |

Reviewer: ```suggestion
| name          | String        | true     | -          | It defines which reducer will be used and will be used as `path` if none is provided:  ex: ` baseUrl/name |
```

| render        | Func          | true     | -          | A function returning a React Node. Ex: ({ list }, props) -> < JSX />                                    |
| renderError   | Func          | false    | -          | A function returning a React Node. Ex: ({ error, creating = bool, error }, props) -> < JSX />           |
| renderLoading | Func          | false    | -          | A function returning a React Node. Ex: ({ create, creating = bool, error }, props) -> < JSX />          |

> | parentId      | String/Number | false    | -          | Parent element's Id, for more specificity in changes.                                                   |

Reviewer: ```suggestion
| parentId      | String/Number | false    | -          | Parent element's ID, for nested resources.                                                   |
```

| path          | String        | false    | `/${name}` | Ex: /foo/bar-\_12?&=                                                                                    |

## How to use

```javascript
import { List } from 'croods'

export default (
  <List
    name="pets"
    // if path is different than /pets, you can set a custom path like so:
    // path='/customPath'
    render={list => <Component list={list} />}
  />
)
```

### render

> The render function is used to receive the api response in the form of an array, **list** , and

Reviewer: ```suggestion
The render function is used to receive the API response as a collection/array to be used to map through and render your resource's component.
```

> pass it to any component you like.

Reviewer: ```suggestion
```

### renderError

The renderError function is fired when the server response has a 4xx or 5xx status.
It receives the `error` parameter that contains the server error message.

> ### renderError

Reviewer: ```suggestion
### renderLoading
```
